src/py/flwr/server/superlink/state/in_memory_state.py

```diff
@@ -205,23 +210,46 @@ def num_task_res(self) -> int:
         """
         return len(self.task_res_store)
 
-    def create_node(self, ping_interval: float) -> int:
+    def create_node(
+        self, ping_interval: float, public_key: Optional[bytes] = None
+    ) -> int:
         """Create, store in state, and return `node_id`."""
         # Sample a random int64 as node_id
         node_id: int = int.from_bytes(os.urandom(8), "little", signed=True)
 
         with self.lock:
-            if node_id not in self.node_ids:
-                self.node_ids[node_id] = (time.time() + ping_interval, ping_interval)
-                return node_id
-        log(ERROR, "Unexpected node registration failure.")
-        return 0
+            if node_id in self.node_ids:
+                log(ERROR, "Unexpected node registration failure.")
+                return 0
 
-    def delete_node(self, node_id: int) -> None:
+            if public_key is not None:
+                if (
+                    public_key in self.public_key_to_node_id
+                    or self.public_key_to_node_id.get(public_key) == node_id
```

Comment:
This line seems to be unreachable. If the part after the `or` is true, doesn't it imply that the part before `or` must also be true?